aiida/tools/visualization/graph.py

```diff
@@ -0,0 +1,777 @@
+            "penwidth": 0
+        }
+
+    node_type_map = {
+        "data.code.Code.": {
+            "shape": "ellipse",
+            "style": "filled",
+            "fillcolor": "#4ca4b9aa",  # blue
+            "penwidth": 0
+        },
+        "process.calculation.calcjob.CalcJobNode.": {
+            "shape": "rectangle",
+            "style": "filled",
+            "fillcolor": "#de707fff",  # red
+            "penwidth": 0
+        },
+        "process.calculation.calcfunction.CalcFunctionNode.": {
+            "shape": "rectangle",
+            "style": "filled",
+            "fillcolor": "#de707f77",  # red
+            "penwidth": 0
+        },
+        "process.workflow.workchain.WorkChainNode.": {
+            "shape": "rectangle",
+            "style": "filled",
+            "fillcolor": "#e38851ff",  # orange
+            "penwidth": 0
+        },
+        "process.workflow.workfunction.WorkFunctionNode.": {
+            "shape": "rectangle",
+            "style": "filled",
+            "fillcolor": "#e38851ff",  # orange
+            "penwidth": 0
+        }
+    }
+
+    node_style = node_type_map.get(class_node_type, default)
+
+    return node_style
+
+
+def pstate_node_styles(node):
+    """map a process node to a graphviz node style
+
+    :param node: the node to map
+    :type node: aiida.orm.nodes.node.Node
+    :rtype: dict
+
+    """
+    class_node_type = node.class_node_type
+
+    default = {"shape": "rectangle", "pencolor": "black"}
+
+    process_map = {
+        "process.calculation.calcjob.CalcJobNode.": {
+            "shape": "ellipse",
+            "style": "filled",
+            # "pencolor": "black",
+            "penwidth": 0,
+            "fillcolor": "#ffffffff"
+        },
+        "process.calculation.calcfunction.CalcFunctionNode.": {
+            "shape": "ellipse",
+            "style": "filled",
+            # "pencolor": "black",
+            "penwidth": 0,
+            "fillcolor": "#ffffffff"
+        },
+        "process.workflow.workchain.WorkChainNode.": {
+            "shape": "polygon",
+            "sides": "6",
+            "style": "filled",
+            # "pencolor": "black",
+            "penwidth": 0,
+            "fillcolor": "#ffffffff"
+        },
+        "process.workflow.workfunction.WorkFunctionNode.": {
+            "shape": "polygon",
+            "sides": "6",
+            "style": "filled",
+            # "pencolor": "black",
+            "penwidth": 0,
+            "fillcolor": "#ffffffff"
+        }
+    }
+
+    node_style = process_map.get(class_node_type, default)
+
+    if isinstance(node, ProcessNode):
+        # style process node, based on success/failure of process
+        if node.is_failed or node.is_excepted or node.is_killed:
+            node_style['fillcolor'] = '#de707fff'  # red
+        elif node.is_finished_ok:
+            node_style['fillcolor'] = '#8cd499ff'  # green
+        else:  # specifically look for waiting state?
```

Comment:
Note that this conditional will hit the states `CREATED` and `RUNNING` as well. Think that the same color as `WAITING` is fine, just thought I'd mention it. Maybe update the comment to include them

Reply:
fixed in 6e599e9